### DISCUSSION TITLE
Database backup

### DISCUSSION THREAD

**Discussion by math-silva** [vercel/community#3002] [Help] [answered]
### Summary

I have reached my usage limits for my database using the Hobby plan, so I can't access it anymore. How can I get a backup of my database?

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/51101359/29282b3a-ec67-485b-ae5a-0461a6d109a3)

**Comment by oliverlevay** [comment]
I have this problem as well

**Comment by SteveChurch** [comment]
You have to do this yourself using 'pg_dump' within a function and store it somewhere of your choice when you have access again.

**Reply by hiraaziz** [reply]
I have postgress database on vercel. How to use pg_dump to get backup of database. Where I have to run this command.

**Comment by z4nr34l** [comment, marked as answer]
You can use `pg_dump` and `pg_restore` to export and restore data: [vercel.com/docs/storage/vercel-postgres/faq#external-data-migration](https://vercel.com/docs/storage/vercel-postgres/faq#external-data-migration)

**Reply by paramaggarwal** [reply]
This does not work by the time the database is suspended due to quota limits exceeded.

**Comment by paramaggarwal** [comment]
Once we reach the limit, the `pg_dump` command does not work. This is a large trust gap on Vercel Postgres database right now. They can lock you out if you don't pay.

**Comment by paramaggarwal** [comment]
Ran into this issue with Vercel Postgres and Vercel Blob Store. In both cases once the limit is reached, the data is held hostage. I think there should be a way to take a data export and delete it.

**Comment by SakalauskasLT** [comment]
<img width="1272" alt="Screenshot 2024-05-10 at 13 25 07" src="https://github.com/vercel/community/assets/74143676/8a135c08-33cc-48fd-9e04-d2eed7aaf3c2">
I have same issue and I didnt even see how I hit those limits

**Comment by SakalauskasLT** [comment]
Now I see Storage size is rising and now 258MB, even I have no access to db. Can I somehow drop some tables to extract all data and move out or this is permanent?

I even see that there is a limit noted of 512MB, not 256MB:

<img width="869" alt="Screenshot 2024-05-15 at 16 15 25" src="https://github.com/vercel/community/assets/74143676/488c50a7-b117-444b-9110-d0c5f597b1ae">
